File: 2610-convert-an-array-into-a-2d-array-with-conditions/2610-convert-an-array-into-a-2d-array-with-conditions.java
class Solution {
    public List<List<Integer>> findMatrix(int[] nums) {
        int freq[] = new int [nums.length + 1];
        // System.out.println(freq.length);
        
        ArrayList<List<Integer>> ans = new ArrayList<>();
        for (int c : nums) {
            System.out.println("freq[c]: " + freq[c]);
            System.out.println("ans size: " + ans.size());
            if (freq[c] >= ans.size()) {
                ans.add(new ArrayList<>());
            }
            
            // add the element to the ans array and then increment the freq of the element
            ans.get(freq[c]).add(c);
            freq[c]++;
        }
        return ans;
    }
}

File: Algorithms-Java/2610-convert-an-array-into-a-2d-array-with-conditions/2610-convert-an-array-into-a-2d-array-with-conditions.js
/**
 * @param {number[]} nums
 * @return {number[][]}
 */
var findMatrix = function(nums) {
    
    let freq = new Array(nums.length + 1).fill(0);
    let ans = [];
    // console.log(freq);
    
    for (let n of nums) {
        if(freq[n] >= ans.length) {
            ans.push([]);
        }
        
        ans[freq[n]].push(n);
        freq[n]++;
    }
    
    return ans;
    
};